dagshub/data_engine/dtypes.py

```diff
@@ -0,0 +1,37 @@
+import enum
+
+class ReservedTags(enum.Enum):
+    ANNOTATION = "annotation"
+
+class DagshubDataType:
+    def __init__(self, value):
+        self.tags = []
+        self.value = value
+
+    def as_annotation(self):
+        self.tags.append(ReservedTags.ANNOTATION.value)
+        return self
+
+    def is_annotation(self):
+        return ReservedTags.ANNOTATION in self.tags
+
+    def tag(self, tag):
+        if tag not in ReservedTags._value2member_map_:
+            self.tags.append(tag)
+            return self
+        raise ValueError("This tag is reserved, choose a different tag name")
```

Comment:
Very bad idea to rely on private members like that. 
Can't guarantee it won't disappear at some point, so only do this type of stuff as a last resort when you are out of other options

You can do the same thing here by doing a test assignment:
```python
try:
  val = ReservedTags(tag)
  is_reserved = True
except ValueError:
  is_reserved = False
```